mne_bids/tests/test_read.py

```diff
@@ -196,17 +199,26 @@ def test_handle_info_reading():
     raw = mne.io.read_raw_fif(raw_fname)
     raw.info['line_freq'] = 60
 
+    # write copy of raw with line freq of 60
     # bids basename and fname
     bids_basename = make_bids_basename(subject='01', session='01',
                                        task='audiovisual', run='01')
     kind = "meg"
     bids_fname = bids_basename + '_{}.fif'.format(kind)
+    write_raw_bids(raw, bids_basename, bids_root, overwrite=True)
 
     # find sidecar JSON fname
-    write_raw_bids(raw, bids_basename, bids_root, overwrite=True)
     sidecar_fname = _find_matching_sidecar(bids_fname, bids_root,
                                            '{}.json'.format(kind),
                                            allow_fail=True)
+    # make a copy of the sidecar in "derivatives/"
+    # to check that we make sure we always get the right sidecar
+    deriv_dir = op.join(bids_root, "derivatives")
+    sidecar_copy = op.join(deriv_dir, op.basename(sidecar_fname))
+    os.mkdir(deriv_dir)
+    with open(sidecar_fname, "r") as fin:
+        sidecar_json = json.load(fin)
+    _write_json(sidecar_copy, sidecar_json)
```

Comment:
will this test fail on master? I don't think so ...

Reply:
you should write a different frequency or update the side car to ensure that the test fails on master

Reply:
```
================================================================== FAILURES ==================================================================
__________________________________________________________ test_handle_info_reading __________________________________________________________

    def test_handle_info_reading():
        """Test reading information from a BIDS sidecar.json file."""
        bids_root = _TempDir()
    
        # read in USA dataset, so it should find 50 Hz
        raw = mne.io.read_raw_fif(raw_fname)
        raw.info['line_freq'] = 60
    
        # bids basename and fname
        bids_basename = make_bids_basename(subject='01', session='01',
                                           task='audiovisual', run='01')
        kind = "meg"
        bids_fname = bids_basename + '_{}.fif'.format(kind)
    
        # find sidecar JSON fname
        write_raw_bids(raw, bids_basename, bids_root, overwrite=True)
        sidecar_fname = _find_matching_sidecar(bids_fname, bids_root,
                                               '{}.json'.format(kind),
                                               allow_fail=True)
    
        # make a copy of the sidecar in "derivatives/"
        # to check that we make sure we always get the right sidecar
        deriv_dir = op.join(bids_root, "derivatives")
        import os
        import json
        from mne_bids.utils import _write_json
        sidecar_copy = op.join(deriv_dir, op.basename(sidecar_fname))
        os.mkdir(deriv_dir)
        with open(sidecar_fname, "r") as fin:
            sidecar_json = json.load(fin)
            sidecar_json["PowerLineFrequency"] = "nonsense"
        _write_json(sidecar_copy, sidecar_json)
    
        # assert that we get the same line frequency set
        raw = mne_bids.read_raw_bids(bids_fname, bids_root)
        assert raw.info['line_freq'] == 60
    
        # 2. if line frequency is not set in raw file, then default to sidecar
        raw.info['line_freq'] = None
        write_raw_bids(raw, bids_basename, bids_root, overwrite=True)
        _update_sidecar(sidecar_fname, "PowerLineFrequency", 55)
        raw = mne_bids.read_raw_bids(bids_fname, bids_root)
>       assert raw.info['line_freq'] == 55
E       assert None == 55
```

I ran this copy on master branch and it fails.

Reply:
The reason it fails is because if you write the same type of file into a sub-directory that is "allowed" in BIDS (e.g. derivatives, or sourcedata), then the current `_find_matching_sidecar` will not work because it starts from bids_root and recursively searches all folders. 

In reality, we shouldn't do that since you only need to search recursively the folders within `sub-<id>`. This would then prevent you from doing `_find_matching_sidecar` on participants.tsv/.json but I think at that point, it's sufficient to do something like `Path(bids_root / 'participants.tsv')` to get your path directly.

Reply:
But that doesn't look the same as what you have here, no? You have this extra line:

sidecar_json["PowerLineFrequency"] = "nonsense"

it's not clear to me from the diff why this code should fail on master.

Reply:
also I think it's a bit of an anti-pattern if the test fails not where it is supposed to.